src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/MutationUtils.cs

```diff
@@ -47,5 +45,89 @@ public static DValue<RecordValue> MergeRecords(IEnumerable<FormulaValue> records
 
             return DValue<RecordValue>.Of(FormulaValue.NewRecordFromFields(mergedFields.Select(kvp => new NamedValue(kvp.Key, kvp.Value))));
         }
+
+        public static async Task<FormulaValue> RemoveCore(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            FormulaValue arg0;
+
+            if (args[0] is LambdaFormulaValue arg0lazy)
+            {
+                arg0 = await arg0lazy.EvalAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                arg0 = args[0];
+            }
+
+            if (arg0 is BlankValue || arg0 is ErrorValue)
+            {
+                return arg0;
+            }
+
+            // If any of the argN (N>0) is error, return the error.
+            foreach (var arg in args.Skip(1))
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                if (arg is ErrorValue)
+                {
+                    return arg;
+                }
+
+                if (arg is TableValue tableValue)
+                {
+                    var errorRecord = tableValue.Rows.FirstOrDefault(row => row.IsError);
+                    if (errorRecord != null)
+                    {
+                        return errorRecord.Error;
+                    }
+                }
```

Comment:
we should discuss this. 

This is eagerly scanning the entire table. Can we confirm that's the language semantics?
The danger is:
- that could be a lot of work upfront
- that could prevent future optimizations 
- in paging/delegation cases, we don't download entire table . It could be a random set of rows. If there are error rows, they may or may not show up in that random set.